kb/development/deprecation.md

```diff
@@ -1,22 +1,25 @@
 # Deprecations
 
-To ensure backwards-compatibility between versions of Mbed TLS, functions in the public interface may not be removed except during the release of a new major version. Instead, functions may be 'deprecated', indicating to users that they will be removed in the future.
+To ensure backwards-compatibility between versions of Mbed TLS, functions or constants in the public interface may not be removed except during the release of a new major version. Instead, they may be 'deprecated', indicating to users that they will be removed in the future.
```

Comment:
Nitpick: if we're going to list them all, it should be something like “elements of the public interface (functions, constants, types, etc.)”.

Reply:
I've done this for the first mention but then kept functions and constants elsewhere, since we don't actually document how to deprecate anything else.